tests/Resources/app.js

```diff
@@ -0,0 +1,336 @@
+ * This should consolidate tests together under our API names like 'Ti.Buffer', with subsuites' tests
+ * just represented as separate tests (the sub-suite name gets prefixed to the test name)
+ * @param  {string[]} suites  stack of suite names
+ * @param  {string} testTitle single test name
+ * @return {object}
+ */
+function suiteAndTitle(suites, testTitle) {
+	var i;
+	var char;
+	var index = -1;
+	var suiteName = '';
+	var newTestTitle = '';
+	for (i = 0; i < suites.length; i++) {
+		char = suites[i].charAt(0);
+		if (char === '.' || char === '#') {
+			index = i;
+			break;
+		}
+	}
+	if (index !== -1) {
+		suiteName = suites.slice(0, index).join('.');
+		newTestTitle = suites.slice(index).join(' ') + ' ' + testTitle;
+	} else {
+		suiteName = suites.join('.');
+		newTestTitle = testTitle;
+	}
+	return {
+		suite: suiteName,
+		title: newTestTitle
+	};
+}
+
+// add a special mocha reporter that will time each test run using
+// our microsecond timer
+function $Reporter(runner) {
+	var started,
+		suites = [];
+
+	runner.on('suite', function (suite) {
+		if (suite.title) {
+			suites.push(suite.title);
+		}
+	});
+
+	runner.on('suite end', function (suite) {
+		if (suite.title) {
+			suites.pop();
+		}
+	});
+
+	runner.on('test', function (test) {
+		Ti.API.info('!TEST_START: ' + test.title);
+		started = new Date().getTime();
+	});
+
+	runner.on('pending', function () {
+		// TODO Spit out something like !TEST_SKIP:  ?
+		started = new Date().getTime(); // reset timer. pending/skipped tests basically start and end immediately
+	});
+
+	// 'pending' hook for skipped tests? Does 'pending', then immediate 'test end'. No 'test' event
+
+	runner.on('fail', function (test, err) {
+		test.err = err;
+		failed = true;
+	});
+
+	runner.on('test end', function (test) {
+		const tdiff = new Date().getTime() - started;
+		const fixedNames = suiteAndTitle(suites, test.title);
+		const result = {
+			state: test.state || 'skipped',
+			duration: tdiff,
+			suite: fixedNames.suite,
+			title: fixedNames.title,
+			error: test.err,
+			message: ''
+		};
+
+		if (test.err) {
+			let message = test.err.message || 'Error';
+			let stack = test.err.stack || '';
+			// uncaught
+			if (test.err.uncaught) {
+				message = 'Uncaught ' + message;
+			}
+
+			// if there's both a stack and message property and they differ, but message is in the stack, strip it from the stack
+			if (test.err.message && test.err.stack && stack.indexOf(message) !== -1) {
+				// stack trace includes the message in it!
+				const index = stack.indexOf(message) + message.length;
+				// indent stack trace without msg
+				stack = stack.slice(index ? index + 1 : index).replace(/^/gm, '  ');
+			}
+			result.message = message;
+			result.stack = stack;
+		}
+
+		// Hack around cycles in structure!
+		const seen = [];
+		let stringified = JSON.stringify(result, (key, val) => {
+		   if (val != null && typeof val === 'object') { // eslint-disable-line eqeqeq
```

Comment:
<!--
  2 failure:  tests/Resources/a..., tests/Resources/a...
  0 warning: 
  
  
  DangerID: danger-id-default;
  File: tests/Resources/app.js;
  Line: 284;
-->
- :no_entry_sign: tests/Resources/app.js line 284 – Mixed spaces and tabs. (no-mixed-spaces-and-tabs)
- :no_entry_sign: tests/Resources/app.js line 284 – Use '===' to compare with null. (no-eq-null)